src/api/entities/confidential/ConfidentialAccount/index.ts

```diff
@@ -313,4 +320,57 @@ export class ConfidentialAccount extends Entity<UniqueIdentifiers, string> {
       count,
     };
   }
+
+  /**
+   * Retrieve the ConfidentialTransactions associated to this Account
+   *
+   * @note uses the middlewareV2
+   * @param filters.eventId - the type of transaction (AccountDepositIncoming/AccountDeposit/AccountWithdraw)
+   * @param filters.assetId - the assetId for which the transactions were made
+   * @param filters.size - page size
+   * @param filters.start - page offset
+   */
+  public async getTransactionHistory(
+    filters: Omit<ConfidentialAssetHistoryByConfidentialAccountArgs, 'accountId'> & {
+      size?: BigNumber;
+      start?: BigNumber;
+    }
+  ): Promise<ResultSet<ConfidentialAssetHistoryEntry>> {
+    const { context, publicKey } = this;
+    const { size, start, ...rest } = filters;
+
+    const {
+      data: {
+        confidentialAssetHistories: { nodes, totalCount },
+      },
+    } = await context.queryMiddleware<Ensured<Query, 'confidentialAssetHistories'>>(
+      getConfidentialAssetHistoryByConfidentialAccountQuery(
+        { accountId: publicKey, ...rest },
+        size,
+        start
+      )
+    );
+
+    const data: ConfidentialAssetHistoryEntry[] = nodes.map(
+      ({ id, assetId, transactionId, amount, eventId }) => {
+        return {
+          id,
+          asset: new ConfidentialAsset({ id: assetId }, context),
+          ...(transactionId && {
+            transaction: new ConfidentialTransaction({ id: new BigNumber(transactionId) }, context),
+          }),
```

Comment:
what is this doing? If transactionId is falsy wouldn't this cause a crash? I am assuming this is a work around from where we didn't add in the not null constraint in SQ.

maybe filter for where transactionId is not null instead? Or at least leave a comment explaining how we know this is safe.